tests/components/webostv/test_media_player.py

```diff
@@ -482,35 +482,44 @@ async def test_client_key_update_on_connect(
     assert config_entry.data[CONF_CLIENT_SECRET] == client.client_key
 
 
+@pytest.mark.parametrize(
+    ("is_on", "exception", "error_message"),
+    [
+        (
+            True,
+            WebOsTvCommandError("Some error"),
+            f"Communication error while calling async_media_play for device {TV_NAME}: Some error",
+        ),
+        (
+            True,
+            WebOsTvCommandError("Some other error"),
+            f"Communication error while calling async_media_play for device {TV_NAME}: Some other error",
+        ),
+        (
+            False,
+            None,
+            f"Error calling async_media_play for device {TV_NAME}: Device is off and cannot be controlled",
+        ),
+    ],
+)
 async def test_control_error_handling(
-    hass: HomeAssistant, client, caplog: pytest.LogCaptureFixture
+    hass: HomeAssistant,
+    client,
+    is_on: bool,
+    exception: Exception,
+    error_message: str,
 ) -> None:
     """Test control errors handling."""
     await setup_webostv(hass)
-    client.play.side_effect = WebOsTvCommandError
-    data = {ATTR_ENTITY_ID: ENTITY_ID}
+    client.play.side_effect = exception
+    client.is_on = is_on
+    await client.mock_state_update()
 
-    # Device on, raise HomeAssistantError
-    with pytest.raises(HomeAssistantError) as exc:
+    data = {ATTR_ENTITY_ID: ENTITY_ID}
+    with pytest.raises(HomeAssistantError, match=error_message):
         await hass.services.async_call(MP_DOMAIN, SERVICE_MEDIA_PLAY, data, True)
 
-    assert (
-        str(exc.value)
-        == f"Error calling async_media_play on entity {ENTITY_ID}, state:on"
-    )
-    assert client.play.call_count == 1
-
-    # Device off, log a warning
-    client.is_on = False
-    client.play.side_effect = TimeoutError
-    await client.mock_state_update()
-    await hass.services.async_call(MP_DOMAIN, SERVICE_MEDIA_PLAY, data, True)
-
-    assert client.play.call_count == 2
-    assert (
-        f"Error calling async_media_play on entity {ENTITY_ID}, state:off, error:"
-        " TimeoutError()" in caplog.text
-    )
+    assert client.play.call_count == (1 if is_on else 0)
```

Comment:
```suggestion
    assert client.play.call_count == int(is_on)
```
We rather avoid if statements in tests